### PR TITLE
Implemented release notes in XML, JSON, and Protobuf

### PR DESCRIPTION
**Pull request by stevespringett**
See #87

### CODE REVIEW

**Comment by stevespringett** [comment]
TODO: Add doc enhancement for attachment text to describe proactive controls

**Comment by stevespringett** [comment]
TODO: Add support for Properties to release notes

**Comment by stevespringett** [comment]
@coderpatros both recommendations have been implemented and include:
* Add doc enhancement for attachment text to describe proactive controls
* Add support for Properties to release notes

**Review by coderpatros — APPROVED** [review]
LGTM +1

File: schema/bom-1.4-SNAPSHOT.xsd [hunk 1]
```diff
@@ -1501,6 +1516,171 @@ limitations under the License.
...
+                <xs:annotation>
+                    <xs:documentation>The URL to an image that may be used in messaging on social media platforms.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="description" type="xs:string" minOccurs="0" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>A short description of the release.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="timestamp" type="xs:dateTime" minOccurs="0">
```
Comment (DarthHater) [inline schema/bom-1.4-SNAPSHOT.xsd:1595]: Pretty sure this should be `maxOccurs="1"` too, yeah?

File: tools/src/test/resources/1.4/valid-release-notes-1.4.json [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+              "contentType": "text/html",
+              "encoding": "base64",
+              "content": "PGgxPk15IG5ldyByZWxlYXNlPGgxPgo8cD5Ob3RhcyBkZSBsYSB2ZXJzacOzbiBhcXXDrTwvcD4="
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
```
Comment (DarthHater) [inline tools/src/test/resources/1.4/valid-release-notes-1.4.json:194]: Trivial: newline

File: schema/bom-1.4-SNAPSHOT.proto [hunk 3]
```diff
@@ -450,3 +456,48 @@ message Evidence {
...
+  RELEASE_TYPE_MAJOR = 0;
+  // A minor release, also known as an update, may contain a smaller number of changes than major releases.
+  RELEASE_TYPE_MINOR = 1;
+  // Patch releases are typically unplanned and may resolve defects or important security issues.
+  RELEASE_TYPE_PATCH = 2;
+  // A pre-release may include alpha, beta, or release candidates and typically have limited support. They provide the ability to preview a release prior to its general availability.
+  RELEASE_TYPE_PRE_RELEASE = 3;
+  // Internal releases are not for public consumption and are intended to be used exclusively by the project or manufacturer that produced it.
+  RELEASE_TYPE_INTERNAL = 4;
+}
```
Comment (DarthHater) [inline schema/bom-1.4-SNAPSHOT.proto:471]: I like this idea but I'm a lil concerned it's a bit too much of one size fits all. I know versions are messy, and I think the biggest thing I can't see how I'd represent here, is mostly an `epoch` in the linux package world. I'd struggle to think of how we could best represent this beyond like suggestions (USE SEMVER PLZ) but a lil less strict.

File: schema/bom-1.4-SNAPSHOT.xsd (hunk 1 above)
Comment (stevespringett) [reply schema/bom-1.4-SNAPSHOT.xsd]: Good catch. Resolved.

File: tools/src/test/resources/1.4/valid-release-notes-1.4.json (hunk 2 above)
Comment (stevespringett) [reply tools/src/test/resources/1.4/valid-release-notes-1.4.json]: Resolved in efed5c41ca8653629f1795084f0fe24078919eca

File: schema/bom-1.4-SNAPSHOT.proto (hunk 3 above)
Comment (stevespringett) [reply schema/bom-1.4-SNAPSHOT.proto]: Resolved in 2e8eca48ba20b431e05e53452c29806bd672e6ec. I've changed the release type from enum to a string, and added descriptions on recommended values.

**Comment by stevespringett** [comment]
@DarthHater Can I get approval on this please?

**Comment by stevespringett** [comment]
@DarthHater Can I get approval on this please?

**Pull request merged**